src/Workspaces/Core/Portable/Shared/Utilities/IWorkspaceThreadingService.cs

```diff
@@ -26,4 +27,9 @@ internal interface IWorkspaceThreadingService
     {
         TResult Run<TResult>(Func<Task<TResult>> asyncMethod);
     }
+
+    internal interface IWorkspaceThreadingServiceProvider : IWorkspaceService
+    {
+        IWorkspaceThreadingService Service { get; }
+    }
```

Comment:
moral equivalent of IWorkspaceThreadingService @jasonmalinowski .  However the above one is only available through MEF, where i need it when i only have a workspace available.